core/src/main/java/com/datastrato/gravitino/authorization/AuthorizationUtils.java

```diff
@@ -69,4 +69,40 @@ public static Namespace ofGroupNamespace(String metalake) {
   public static Namespace ofUserNamespace(String metalake) {
     return Namespace.of(metalake, Entity.SYSTEM_CATALOG_RESERVED_NAME, Entity.USER_SCHEMA_NAME);
   }
+
+  public static void checkUser(NameIdentifier ident) {
+    NameIdentifier.check(ident != null, "User identifier must not be null");
+    checkUserNamespace(ident.namespace());
+  }
+
+  public static void checkGroup(NameIdentifier ident) {
+    NameIdentifier.check(ident != null, "Group identifier must not be null");
+    checkGroupNamespace(ident.namespace());
+  }
+
+  public static void checkRole(NameIdentifier ident) {
+    NameIdentifier.check(ident != null, "Role identifier must not be null");
+    checkRoleNamespace(ident.namespace());
+  }
+
+  public static void checkUserNamespace(Namespace namespace) {
+    Namespace.check(
+        namespace != null && namespace.length() == 3,
+        "User namespace must be non-null and have 3 levels, the input namespace is %s",
```

Comment:
`User namespace must have 3 levels`

Reply:
ditto

Reply:
It is a little weird. 
Because `have three levels` include the case `it's non-null`.